src/main/java/com/powsybl/openloadflow/sensi/AbstractSensitivityAnalysis.java

```diff
@@ -49,217 +47,294 @@ protected AbstractSensitivityAnalysis(MatrixFactory matrixFactory, Supplier<Grap
         this.connectivityProvider = Objects.requireNonNull(connectivityProvider);
     }
 
+    protected static Terminal getEquipmentRegulatingTerminal(Network network, String equipmentId) {
+        Generator generator = network.getGenerator(equipmentId);
+        if (generator != null) {
+            return generator.getRegulatingTerminal();
+        }
+        StaticVarCompensator staticVarCompensator = network.getStaticVarCompensator(equipmentId);
+        if (staticVarCompensator != null) {
+            return staticVarCompensator.getRegulatingTerminal();
+        }
+        TwoWindingsTransformer t2wt = network.getTwoWindingsTransformer(equipmentId);
```

Comment:
This PR is indeed not working with transformers as we don't have any PV bus linked to the transformer regulation. In your test, the PV bus is linked to the generator. For phase tap changers, we have activated the phase variable. Here, we have to active the rho variable and the linked bus_v equation I think.